test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientIntegrationTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿// <copyright file="SqlClientIntegrationTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Data;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using DotNet.Testcontainers.Containers;
+using Microsoft.Data.SqlClient;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+using Testcontainers.MsSql;
+using Testcontainers.SqlEdge;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.SqlClient.Tests
+{
+    public sealed class SqlClientIntegrationTests : IAsyncLifetime
+    {
+        // The Microsoft SQL Server Docker image is not compatible with ARM devices, such as Macs with Apple Silicon.
+        private readonly IContainer databaseContainer = Architecture.Arm64.Equals(RuntimeInformation.ProcessArchitecture) ? new SqlEdgeBuilder().Build() : new MsSqlBuilder().Build();
+
+        public Task InitializeAsync()
+        {
+            return this.databaseContainer.StartAsync();
+        }
+
+        public Task DisposeAsync()
+        {
+            return this.databaseContainer.DisposeAsync().AsTask();
+        }
+
+        [Trait("CategoryName", "SqlIntegrationTests")]
```

Comment:
Doesn't need to be in this PR, but this test extends our build/test CI by about 1 minute. This will only get longer as we port our other integration tests to use TestContainers.

We could add filter like `dotnet test --filter Traits\!~IntegrationTests` here to exclude them:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/094124e9ee61466a5bbf182d923658cde2c2dcf5/.github/workflows/ci.yml#L41-L42

Then we could make a separate integration test job that just ran all tests with trait equal to `*IntegrationTests`

Reply:
Usually, putting tests in different assemblies and running them parallel can improve performance. In some cases, running tests against a singleton instance of a container or parallel against multiple instances can also increase performance, although this depends on the specific test. However, at the moment, I believe that the additional test time will be mostly due to pulling the image.